schema/deploy/shipping/views.sql

```diff
@@ -2911,4 +2913,198 @@ comment on view shipping.uw_reopening_ehs_reporting_v1 is
+            (select case string_response[1] when 'yes' then true else false end from shipping.fhir_questionnaire_responses_v1 where encounter_id = all_uw_instances.encounter_id and link_id = 'daily_exposure_known_pos') as daily_exposure_known_pos,
+            (select string_response[1] from shipping.fhir_questionnaire_responses_v1 where encounter_id = all_uw_instances.encounter_id and link_id = 'testing_trigger') as testing_trigger,
+            (select string_response[1] from shipping.fhir_questionnaire_responses_v1 where encounter_id = all_uw_instances.encounter_id and link_id = 'surge_selected') as surge_selected,
+            sample.collected::date as sample_collection_date
+        from all_uw_instances
+        left join warehouse.sample using (encounter_id)
+        where redcap_event_name = 'encounter_arm_1'
+    ),
+
+    uw_individuals as (
+        select
+            individual,
+            max(uw_encounters.encountered) filter (where testing_trigger = '1') as latest_invite_date,
+            max(sample_collection_date) filter (where sample_collection_date is not null) as latest_collection_date
+        from uw_enrollments
+        left join uw_encounters using (individual)
+        group by individual
+    ),
+
+    -- Select encounters for testing based on positive daily attestations
+    positive_daily_attestations as (
+        select
+            uw_encounters.redcap_project_id,
+            uw_encounters.redcap_record_id,
+            uw_encounters.redcap_event_name,
+            uw_encounters.redcap_repeat_instance,
+            uw_encounters.encountered,
+            individual,
+            tier,
+            latest_invite_date,
+            latest_collection_date,
+            case
+                when daily_symptoms then 1
+                when daily_exposure_known_pos then 2
+                when daily_exposure then 3
+                else null
+            end as priority,
+            case
+                when daily_symptoms then 'symptomatic'
+                when daily_exposure_known_pos then 'exposure_to_known_positive'
+                when daily_exposure then 'gathering_over_10'
+                else null
+            end as priority_reason
+        from uw_encounters
+        join uw_individuals using (individual)
+        join uw_enrollments using (individual)
+        -- Filter to encounters for participants whose last invite was over 7 days before encounter
+        where (latest_invite_date is null or latest_invite_date < uw_encounters.encountered - interval '7 days')
+        -- Filter to encounters for participants who have never had a sample collected or their last sample collection was over 7 days before encounter
+        and (latest_collection_date is null or latest_collection_date < uw_encounters.encountered - interval '7 days')
+        -- Filter for instances taht do no already have testing_trigger filled
+        and testing_trigger is null
```

Comment:
I had assumed that we can rely on the REDCap/FHIR ETL to not ingest empty string values. 
We could specifically check `testing_trigger is null or trim(testing_trigger) = ''`

Reply:
Right, the REDCap ETL won't write a questionnaire item if its answer is ''. 
But to be safe if would make sense to check.

Reply:
Instead of sprinkling that empty string check in a few places, it might be a cleaner separation of concerns to use `nullif(…, '') as testing_trigger` in the view which provides the `testing_trigger` column. (And ditto with `surge_selected`.)

Reply:
Updated `testing_trigger` and `surge_selected_flag` to be boolean values.